scripts/delete-federation-with-helm-chart.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script removes the cluster registry, unjoins any clusters passed as
+# arguments, and removes the federation from the current kubectl context.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source "$(dirname "${BASH_SOURCE}")/util.sh"
+
+set -x
+
+KCD="kubectl --ignore-not-found=true delete"
+NS="${FEDERATION_NAMESPACE:-federation-system}"
+PUBLIC_NS=kube-multicluster-public
+NAMESPACED="${NAMESPACED:-}"
+
+KF_NS_ARG="--federation-namespace=${NS} "
+if [[ "${NAMESPACED}" ]]; then
+  KF_NS_ARG+="--registry-namespace=${NS}"
+fi
+
+# Unjoin clusters by removing objects added by kubefed2.
+HOST_CLUSTER="$(kubectl config current-context)"
+JOINED_CLUSTERS="$(kubectl -n "${NS}" get federatedclusters -o=jsonpath='{range .items[*]}{.metadata.name}{" "}{end}')"
+for c in ${JOINED_CLUSTERS}; do
+  ./bin/kubefed2 unjoin "${c}" --host-cluster-context "${HOST_CLUSTER}" --remove-from-registry --v=2 ${KF_NS_ARG}
+done
+
+${KCD} -n "${NS}" FederatedTypeConfig --all
```

Comment:
Currently we have no function to check federated resources before delete federation. I think it should be another function requirement if it is a case.

Reply:
This is the use case, suppose we use the `helm install` install the federation, and then joined some clusters, also created some federated resources. 

Then if I run this script, it will be failed, as the deletion of delete those federated resources will be failed.

How about add some checking before `unjoin` in Line36, if there some federated resources, just report some warning message and return, do not unjoin and delete resources for such cases.

@xunpan @marun